velocidi-sdk/src/main/kotlin/com/velocidi/HttpClient.kt

```diff
@@ -0,0 +1,63 @@
+package com.velocidi
+
+import android.util.Log
+import com.android.volley.RequestQueue
+import com.android.volley.Response
+import com.android.volley.toolbox.*
+import java.io.File
+import com.android.volley.toolbox.StringRequest
+import org.json.JSONObject
+import java.nio.charset.Charset
+
+
+class HttpClient {
+    private val cache = DiskBasedCache(File(Constants.CACHE_DIR), 1024 * 1024) // 1MB cap
+
+    private val network = BasicNetwork(HurlStack())
+
+    val requestQueue = RequestQueue(cache, network).apply {
+        start()
+    }
+
+    val headers = mutableMapOf(Pair("Content-Type", "application/json"))
+    val defaultParams = mutableMapOf<String,String>()
+
+
+    fun sendRequest(verb: Verb,url: String, payload: JSONObject? = null, listener: ResponseListener? = null) {
+        val successListener = Response.Listener<String> {
+                response -> listener?.onResponse(response) ?: Log.i(Constants.LOG_TAG, response) }
+
+        val errorListener = Response.ErrorListener {
+                error -> listener?.onError(error.toString()) ?: Log.i(Constants.LOG_TAG, error.toString()) }
```

Comment:
Does Kotlin have a `Try<T>` ?

If not, I think this interface should provide a `.onError(Exception)` or something of the likes...

Reply:
Done